src/Admin/CitizenProjectAdmin.php

```diff
@@ -277,9 +292,84 @@ protected function configureDatagridFilters(DatagridMapper $datagridMapper)
+            ->add('creatorFirstName', CallbackFilter::class, [
+                'label' => 'Prénom du créateur',
+                'show_filter' => true,
+                'field_type' => TextType::class,
+                'callback' => function (ProxyQuery $qb, string $alias, string $field, array $value) {
+                    if (!$value['value']) {
+                        return;
+                    }
+
+                    /* @var ProxyQuery|QueryBuilder $qb */
+                    $qb
+                        ->leftJoin(Adherent::class, 'creator', Join::WITH, sprintf('creator.uuid=%s.createdBy', $alias))
+                        ->andWhere(
+                            'LOWER(creator.firstName) LIKE :firstName '.
+                            "OR $alias.createdBy=:uuid"
+                        )
+                        ->setParameter('firstName', sprintf('%%%s%%', mb_strtolower($value['value'])))
+                        ->setParameter('uuid', $value['value']);
+
+                    return true;
+                },
+            ])
+            ->add('creatorLastName', CallbackFilter::class, [
+                'label' => 'Nom du créateur',
+                'show_filter' => true,
+                'field_type' => TextType::class,
+                'callback' => function (ProxyQuery $qb, string $alias, string $field, array $value) {
+                    if (!$value['value']) {
+                        return;
+                    }
+
+                    /** @var ProxyQuery|QueryBuilder $qb */
+                    if (!in_array('creator', $qb->getAllAliases(), true)) {
+                        $qb->leftJoin(Adherent::class, 'creator', Join::WITH, sprintf('creator.uuid=%s.createdBy', $alias));
+                    }
+
+                    $qb->andWhere(
+                            'LOWER(creator.lastName) LIKE :lastName '.
+                            "OR $alias.createdBy=:uuid"
+                        )
+                        ->setParameter('lastName', sprintf('%%%s%%', mb_strtolower($value['value'])))
+                        ->setParameter('uuid', $value['value']);
+
+                    return true;
+                },
+            ])
+            ->add('creatorEmailAddress', CallbackFilter::class, [
+                'label' => 'Email du créateur',
+                'show_filter' => true,
+                'field_type' => EmailType::class,
+                'callback' => function (ProxyQuery $qb, string $alias, string $field, array $value) {
+                    if (!$value['value']) {
+                        return;
+                    }
+
+                    /** @var ProxyQuery|QueryBuilder $qb */
+                    if (!in_array('creator', $qb->getAllAliases(), true)) {
+                        $qb->leftJoin(Adherent::class, 'creator', Join::WITH, sprintf('creator.uuid=%s.createdBy', $alias));
+                    }
+
+                    $qb->andWhere(
+                            'LOWER(creator.emailAddress)=:emailAddress '.
```

Comment:
you can avoid `LOWER` function on email field and on `_ci` collate, it's already saved as lowercase string:
```
//MembershipRequest
public function setEmailAddress(string $emailAddress): void
    {
        $this->emailAddress = mb_strtolower($emailAddress);
    }
```

Reply:
Yea great :-)